src/crawlers/puppeteer_crawler.js

```diff
@@ -265,17 +282,28 @@ class PuppeteerCrawler {
      * @ignore
      */
     async _handleRequestFunction({ request, autoscaledPool }) {
+        let session;
         const page = await this.puppeteerPool.newPage();
 
+        if (this.sessionPool) {
+            // @TODO: Handle cookies load here.
+            const browser = page.browser();
+            // @TODO: Session pool method find my id;
+            session = this.sessionPool.sessions.find(ses => ses.id === browser[BROWSER_SESSION_ID_KEY_NAME]);
```

Comment:
yeah this doesn't look good, it's using an internal property plus O(n) search...

Reply:
can't you save the session on the browser object directly, rather then indirectly via id?